pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2153,9 +2153,14 @@ private CompletableFuture<Void> seekAsyncInternal(long requestId, ByteBuf seek,
 
         MessageIdAdv originSeekMessageId = seekMessageId;
         seekMessageId = (MessageIdAdv) seekId;
-        duringSeek.set(true);
+        
+        if (!duringSeek.compareAndSet(false, true)) {
+            log.warn("[{}][{}] Attempting to seek operation that is already in progress, cancelling {}", 
+                    topic, subscription, seekBy);
+            seekFuture.cancel(true);
```

Comment:
Also, this is a local future, that I believe you can complete it properly (@michaeljmarshall shall we complete with `null` or exception here?) instead of call `cancel`.

Reply:
Good catch! I prefer  `seekFuture.completeExceptionally()` instead of `cancel(true);`.

Or we can use a global future instead local future, see https://github.com/apache/pulsar/pull/18202.

Reply:
Sounds reasonable with your test case. I'll give another look tomorrow.

Reply:
After a closer look `f.cancel(true)` is the same as `f.completeExceptionally(new CancellationException())`. So either way should be OK.

Reply:
Nit: I think an `IllegalStateException` would be a clearer exception indicating to the user that it was not a valid time to call this method. One problem with `CancellationException` is that it has no message. However, I see that there is a warning log indicating the problem, so it might be fine.